include/ofxPropsMetadata.h

```diff
@@ -0,0 +1,224 @@
+// Copyright OpenFX and contributors to the OpenFX project.
+// SPDX-License-Identifier: BSD-3-Clause
+// NOTE: This file is auto-generated by gen-props.py. DO NOT EDIT.
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include "ofxImageEffect.h"
+#include "ofxGPURender.h"
+#include "ofxColour.h"
+#include "ofxDrawSuite.h"
+#include "ofxParametricParam.h"
+#include "ofxKeySyms.h"
+#include "ofxOld.h"
+
+namespace OpenFX {
+enum class PropType {
+   Int,
+   Double,
+   Enum,
+   Bool,
+   String,
+   Bytes,
+   Pointer
+};
+
+enum class Writable {
+   Host,
+   Plugin,
+   All
+};
+
+struct PropsMetadata {
+  std::string name;
+  std::vector<PropType> types;
+  int dimension;
+  Writable writable;
+  bool host_optional;
+  std::vector<const char *> values; // for enums
```

Comment:
You should be consistant, if you use `std::string` for name, you should also use `std::vector<std::string>`

Reply:
I've tried now to move to `char *`, to avoid allocations where possible during startup. I should move to `std::string_view` in some places. I'll do that.